handlers/node_api.py

```diff
@@ -60,18 +54,201 @@ def get_address_search(
 
     response = node_api.get_address_search(address, count, page, hash, token)
 
-    if response is None:
-        return {
-            "statusCode": 503,
-            "body": json.dumps({"error": "unknown_error"}),
-            "headers": {
-                "Content-Type": "application/json"
-            }
+    return {
+        "statusCode": 200,
+        "body": json.dumps(response or UNKNOWN_ERROR_MSG),
+        "headers": {
+            "Content-Type": "application/json"
+        }
+    }
+
+
+@ApiGateway()
+def get_version(
+    event: ApiGatewayEvent,
+    _context: LambdaContext,
+    node_api: Optional[NodeApi] = None
+) -> dict:
+    node_api = node_api or NodeApi()
+
+    response = node_api.get_version()
+
+    return {
+        "statusCode": 200,
+        "body": json.dumps(response or UNKNOWN_ERROR_MSG),
+        "headers": {
+            "Content-Type": "application/json"
+        }
+    }
+
+
+@ApiGateway()
+def get_dashboard_tx(
+    event: ApiGatewayEvent,
+    _context: LambdaContext,
+    node_api: Optional[NodeApi] = None
+) -> dict:
+    node_api = node_api or NodeApi()
+
+    block = event.query.get("block")
+    tx = event.query.get("tx")
+
+    if block is None or tx is None:
+        raise ApiError("invalid_parameters")
+
+    response = node_api.get_dashboard_tx(block, tx)
+
+    return {
+        "statusCode": 200,
+        "body": json.dumps(response or UNKNOWN_ERROR_MSG),
+        "headers": {
+            "Content-Type": "application/json"
+        }
+    }
+
+
+@ApiGateway()
+def get_transaction_acc_weight(
+    event: ApiGatewayEvent,
+    _context: LambdaContext,
+    node_api: Optional[NodeApi] = None
+) -> dict:
+    node_api = node_api or NodeApi()
+    id = event.query.get("id")
+
+    if id is None:
+        raise ApiError("invalid_parameters")
+
+    response = node_api.get_transaction_acc_weight(id)
+
+    return {
+        "statusCode": 200,
+        "body": json.dumps(response or UNKNOWN_ERROR_MSG),
+        "headers": {
+            "Content-Type": "application/json"
         }
+    }
 
+
+@ApiGateway()
+def get_token_history(
+    event: ApiGatewayEvent,
+    _context: LambdaContext,
+    node_api: Optional[NodeApi] = None
+) -> dict:
+    node_api = node_api or NodeApi()
+
+    id = event.query.get("id")
+    count = event.query.get("count")
+    hash = event.query.get("hash")
+    page = event.query.get("page")
+    timestamp = event.query.get("timestamp")
+
+    if id is None or count is None:
+        raise ApiError("invalid_parameters")
+
+    if hash is not None and (page is None or timestamp is None):
+        # If hash exists, it"s a paginated request and page is required
+        raise ApiError("invalid_parameters")
```

Comment:
Wouldn't it be better if we just proxy everything to the fullnode and let it perform the validation?

Otherwise we will need to update here every time someone updates the validation rules in hathor-core.

Reply:
Maybe, but filtering out invalid parameters may lessen the burden on the full-node. If this is needless we can remove easily on another PR.

Reply:
Ok, I see your point